### PR TITLE
Fixes #23068 - upgrade to candlepin 2.3

### PR DESCRIPTION
**Pull request by jlsherrill**
This upgrades katello to use candlepin 2.3, which includes these changes:

* Message bus-based event data formats have changed
* The previously deprecated subscriptions list/get endpoint was removed
  * this endpoint used to combine data from marketing products and pools into one endpoint, thus is being replaced with marketing products
  * The list of products, and the stacking id that was previously on the katello Subscription object is now moved to the Pool object since that is where it is in candlepin
  * the cp_product_id is now the cp_id, as its the id in candlepin of the marketing product
* The Subscription create api is deprecated and that has now been moved to use Pool create

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

File: app/lib/actions/katello/product/destroy.rb [hunk 1]
```diff
@@ -55,6 +55,7 @@ def plan(product, options = {})
 
         def finalize
           product = ::Katello::Product.find(input[:product_id])
+          ::Katello::Subscription.where(:cp_id => product.cp_id, :organization_id => product.organization_id).destroy_all
```
Comment (jlsherrill) [inline app/lib/actions/katello/product/destroy.rb]: why was this needed justin?

File: app/lib/katello/http_resource.rb [hunk 2]
```diff
@@ -82,6 +82,7 @@ def get(a_path, headers = {})
         a_path = URI.encode(a_path)
         client = rest_client(Net::HTTP::Get, :get, a_path)
         result = process_response(client.get(headers))
+
```
Comment (jlsherrill) [inline app/lib/katello/http_resource.rb]: why are you adding space?

**Comment by parthaa** [comment]
1. Reg/Unreg, assign/remove pool, compliance ->  check the entitlements update properly
1. Import/Refresh/Delete Manifest
1. Create custom Product, refresh manifest  , delete mani  and make sure the custom subs are ok
1. Reimport manifest and just make sure the Pools and Subs match 
1.  migration -> Upgrade to candlepin and make sure the Stacking Id from Subs is now on the Pool and product
1. make sure ```rake katello:reimport``` imports subs, pools products

**Comment by jlsherrill** [comment]
Why didn't you remove Subscription.get_for_owner?

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

File: app/lib/actions/katello/product/destroy.rb (hunk 1 above)
Comment (jturel) [reply app/lib/actions/katello/product/destroy.rb]: Did you intend to put a `dependent: :destroy` on K::Pool subscriptions?

**Comment by jturel** [comment]
This works very well after quite a bit of testing. One comment on product destroy action..

**Comment by jturel** [comment]
I tested this against candlepin 2.4.0 so I think we are good to go straight to that version (and actually I was advised to by the candlepin team).

**Comment by jlsherrill** [comment]
Yes, they rebuilt the latest 2.3.z as 2.4.0 for complicated reasons :)   It should be the exact same as the latest build i developed against.  Its the one we'll pull in

**Comment by jlsherrill** [comment]
Added that dependent => destroy on  products -> subscriptions

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Comment by parthaa** [comment]
Nice work @jlsherrill ! ACK 

noworthy
